bigtable/client/instance_admin_test.cc

```diff
@@ -422,3 +422,51 @@ TEST_F(InstanceAdminTest, CreateInstancePollReturnsFailure) {
   EXPECT_THROW(future.get(), bigtable::GRpcError);
 }
 #endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+
+/// @test Verify that DeleteInstance works in the positive case.
+TEST_F(InstanceAdminTest, DeleteInstance) {
+  using namespace ::testing;
+  using google::protobuf::Empty;
+  bigtable::InstanceAdmin tested(client_);
+  std::string expected_text = R"""(
+  name: 'projects/the-project/instances/the-instance'
+      )""";
+  auto mock = MockRpcFactory<btproto::DeleteInstanceRequest, Empty>::Create(
+      expected_text);
+  EXPECT_CALL(*client_, DeleteInstance(_, _, _)).WillOnce(Invoke(mock));
+  // After all the setup, make the actual call we want to test.
+  tested.DeleteInstance("the-instance");
+}
+
+/// @test Verify unrecoverable error for DeleteInstance
+TEST_F(InstanceAdminTest, DeleteInstanceUnrecoverableError) {
+  using namespace ::testing;
+  bigtable::InstanceAdmin tested(client_);
+  EXPECT_CALL(*client_, DeleteInstance(_, _, _))
+      .WillRepeatedly(
+          Return(grpc::Status(grpc::StatusCode::PERMISSION_DENIED, "uh oh")));
+// After all the setup, make the actual call we want to test.
+#if GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+  EXPECT_THROW(tested.DeleteInstance("other-instance"), std::exception);
+#else
+  EXPECT_DEATH_IF_SUPPORTED(tested.DeleteInstance("other-instance"),
+                            "exceptions are disabled");
+#endif  // GOOGLE_CLOUD_CPP_HAVE_EXCEPTIONS
+}
+
+/// @test Verify that recoverable error for DeleteInstance
+TEST_F(InstanceAdminTest, DeleteInstanceRecoverableError) {
+  using namespace ::testing;
+  bigtable::InstanceAdmin tested(client_);
+  EXPECT_CALL(*client_, DeleteInstance(_, _, _))
+      .Times(1)
```

Comment:
I think `.Times(1)` and `WillOnce()` is redundant.  Remove the `Times(1)` ?

Reply:
removed Times(1), but WillOnce causes failure if run with -DGOOGLE_CLOUD_CPP_ENABLE_CXX_EXCEPTIONS=no.
The error is
` RUN      ] InstanceAdminTest.DeleteInstanceRecoverableError
/home/manish/git/google-cloud-cpp/bigtable/client/instance_admin_test.cc:466: Failure
Actual function call count doesn't match EXPECT_CALL(*client_, DeleteInstance(_, _, _))...
         Expected: to be called once
           Actual: never called - unsatisfied and active
[  FAILED  ] InstanceAdminTest.DeleteInstanceRecoverableError (150 ms)
`  

I am not sure of the reason behind it, the work around is to use WillRepeatedly instead.

Reply:
Ahh, yes, the usual problem with `EXPECT_DEATH_.*`, it runs in a separate process, so the call is not counted (or visible) to the parent process.  Thanks for checking.